backend-implicaction/src/test/java/com/dynonuggets/refonteimplicaction/controller/JobsPostingControllerTest.java

```diff
@@ -126,4 +131,63 @@ void getJobByIdWithoutJwtShouldBeForbidden() throws Exception {
 
         verify(jobPostingService, times(0)).getJobById(anyLong());
     }
+
+    @Test
+    @WithMockUser
+    void archiveJobShouldChangeStatus() throws Exception {
+        JobPostingDto jobPostingDtoAvailable = JobPostingDto.builder()
+                .id(1L)
+                .createdAt(Instant.now())
+                .description("description")
+                .location("Paris")
+                .keywords("toto,yoyo")
+                .salary("1111$")
+                .status(statusAvailable)
+                .build();
+
+        JobPostingDto jobPostingDtoArchived = JobPostingDto.builder()
+                .id(1L)
+                .createdAt(Instant.now())
+                .description("description")
+                .location("Paris")
+                .keywords("toto,yoyo")
+                .salary("1111$")
+                .status(statusArchived)
+                .build();
+
+        given(jobPostingService.archiveOrUnarchiveJobPosting(jobPostingDtoAvailable.getId())).willReturn(jobPostingDtoArchived);
+
+        mvc.perform(patch(JOBS_BASE_URI + ARCHIVE_JOB_URI, jobPostingDtoAvailable.getId()).contentType(APPLICATION_JSON)).andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id", is(Math.toIntExact(jobPostingDtoAvailable.getId()))))
+                .andExpect(jsonPath("$.description", is(jobPostingDtoAvailable.getDescription())))
+                .andExpect(jsonPath("$.location", is(jobPostingDtoAvailable.getLocation())))
+                .andExpect(jsonPath("$.keywords", is(jobPostingDtoAvailable.getKeywords())))
+                .andExpect(jsonPath("$.salary", is(jobPostingDtoAvailable.getSalary())))
+                .andExpect(jsonPath("$.status.label", is(jobPostingDtoArchived.getStatus().getLabel())));
+
+        verify(jobPostingService, times(1)).archiveOrUnarchiveJobPosting(anyLong());
+    }
+
+    @Test
+    @WithMockUser
+    void archiveJobPostingWhithUnexistingIdShouldThrowException() throws Exception {
+        final long jobId = 123L;
+        final NotFoundException exception = new NotFoundException(String.format(JOB_NOT_FOUND_MESSAGE, jobId));
+        given(jobPostingService.archiveOrUnarchiveJobPosting(anyLong())).willThrow(exception);
+
+        mvc.perform(patch(JOBS_BASE_URI + ARCHIVE_JOB_URI, jobId)).andExpect(status().isNotFound());
+
+        verify(jobPostingService, times(1)).archiveOrUnarchiveJobPosting(anyLong());
+    }
+
+    @Test
+    void archiveJobByIdWithoutJwtShouldBeForbidden() throws Exception {
+        final long jobId = 123L;
+        mvc.perform(patch(JOBS_BASE_URI + ARCHIVE_JOB_URI, jobId).contentType(APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isForbidden());
+
+        verify(jobPostingService, times(0)).archiveOrUnarchiveJobPosting(anyLong());
```

Comment:
```suggestion
        verify(jobPostingService, never()).archiveOrUnarchiveJobPosting(anyLong());
```

Je sais que j'ai mis `times(0)` au début, mais `never()` c'est mieux 😉